### PR TITLE
feat: Add 1st version of agent support with a weather query tool

### PR DESCRIPTION
**Pull request by nkwangleiGIT**
1. Add 1st version of agent support with a weather query tool to use
Sample agent:
```
apiVersion: agent.arcadia.kubeagi.k8s.com.cn/v1alpha1
kind: Agent
metadata:
  name: weather-agent
  namespace: test
spec:
  type: zeroShot
  allowedTools:
  - name: "Weather Query API"
    parms:
      apiKey: <api key to use>
  options:
    maxIterations: 5
```

Add the following elements to your LLM app:
```yaml
  - description: agent llm
    displayName: agent-llm
    name: agent-llm-node
    nextNodeName:
    - agent-node
    ref:
      apiGroup: arcadia.kubeagi.k8s.com.cn
      kind: LLM
      name: qwen-7b-test
  - description: 天气预报 agent 助手
    displayName: agent
    name: agent-node
    nextNodeName:
    - chain-node
    ref:
      apiGroup: agent.arcadia.kubeagi.k8s.com.cn
      kind: Agent
      name: weather-agent
```
3. Increase default qps and burst
4. support to use remote fastchat API for local apiserver debuging

### CODE REVIEW

File: pkg/appruntime/agent/executor.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+			}
+			allowedTools = append(allowedTools, tool)
+		default:
+			return nil, fmt.Errorf("no tool found with name: %s", toolSpec.Name)
+		}
+	}
+
+	// Initialize executor using langchaingo
+	options := agents.WithMaxIterations(instance.Spec.Options.MaxIterations)
+	executor, err := agents.Initialize(llm, allowedTools, agents.ZeroShotReactDescription, options)
```
Comment (Abirdcfly) [inline pkg/appruntime/agent/executor.go:87]: use agents.ZeroShotReactDescription that means we use prompt in https://github.com/tmc/langchaingo/blob/main/agents/mrkl_prompt.go#L12, Should we override this with our prompt?

File: api/app-node/agent/v1alpha1/agent_types.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+	// ConditionedStatus is the current status
+	v1alpha1.ConditionedStatus `json:",inline"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// Agent is the Schema for the Agent API
+type Agent struct {
```
Comment (Abirdcfly) [inline api/app-node/agent/v1alpha1/agent_types.go:82]: that will be

group: agent.arcadia.kubeagi.k8s.com.cn
kind: Agent

Looks a little redundant?

**Review by Abirdcfly — APPROVED** [review]

File: pkg/appruntime/agent/executor.go (hunk 1 above)
Comment (nkwangleiGIT) [reply pkg/appruntime/agent/executor.go:87]: I think we can keep it, as it's working fine for now. We can override it if we have a better one later.

File: api/app-node/agent/v1alpha1/agent_types.go (hunk 2 above)
Comment (nkwangleiGIT) [reply api/app-node/agent/v1alpha1/agent_types.go:82]: will remove agent from group name

File: config/samples/arcadia_v1alpha1_agent.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+apiVersion: agent.arcadia.kubeagi.k8s.com.cn/v1alpha1
+kind: Agent
+metadata:
+  name: weather-agent
+  namespace: test
+spec:
+  type: zeroShot
+  allowedTools:
```
Comment (Abirdcfly) [inline config/samples/arcadia_v1alpha1_agent.yaml:8]: We may need a way for the user to know what tools our code provides just by using crd, for instance:
```
apiVersion: agent.arcadia.kubeagi.k8s.com.cn/v1alpha1
kind: Agent
metadata:
  name: weather-agent
  namespace: test
spec:
  type: zeroShot
  usedTools:
    weather:
      apikey: xxxx
    calculator: {}
    bingSearch:
      apikey: xxx
```

Comment (nkwangleiGIT) [reply config/samples/arcadia_v1alpha1_agent.yaml:8]: yes, we can think about this later, for now ,we can provide a list in our product and let user to select what capabilities/tools they want to use in their Agent.

**Pull request merged**